_RFCs/002-node-scaffolding-split.md

```diff
@@ -0,0 +1,60 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This splits the node scaffolding into two different scaffoldings - one for dynamic applications and one for applications that are compiled to static assets.
+
+If you are a node developer creating an Express application (or any other application that uses the Node.js web server) you should use the scaffolding for dynamic applications.
+
+If you are a node developer creating a React or Angular application (or any other application that is compiled into static assets, then run behind a proxy like Nginx) you should use the scaffolding for static applications.
+
+Developers who are using the current node scaffolding need to migrate to either the dynamic scaffolding or the static scaffolding. The dynamic scaffolding will largely be the same as the current node scaffolding, but users will need to change the name of the scaffolding they specify when initializing habitat in their application repos (and in their habitat/plan.sh files).
+
+The original scaffolding-node will remain on builder but will be deprecated and no longer updated.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Scaffolding-node splits into two scaffoldings - scaffolding-node-dynamic and scaffolding-node-static (these are just placeholder names, they can be something else).
+
+Scaffolding-node-dynamic remains largely the same as the current scaffolding-node.
+
+Scaffolding-node-static will compile a node application into static assets and will run them behind an Nginx proxy (there will be a dependency on core/nginx).
+
+The code around hab plan init would need to be changed to accept either the scaffolding-node-dynamic or scaffolding-node-static scaffoldings as an -s argument.
+
+All tutorials/documentation that refer to scaffolding-node will need to change to refer to either scaffolding-node-dynamic or scaffolding-node-static.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The original intent of the node scaffolding was to be a quickstart for Habitizing any node application. This approach would force node developers to think about which scaffolding would serve them best and choose one or the other.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+The node ecosystem is vast and it is very unlikely we could ever support all common node implementations with one scaffolding.
+
+An alternative would be to include two options with the current node scaffolding - and have the user decide which one to use with an environmental variable. But this would make the current node scaffolding extremely complex and make it more difficult to maintain/contribute to.
```

Comment:
What would that complexity actually look like, though? I haven't spent much time with the scaffolding yet, but in my imagination at least, it seems like the main deviations from current would be adding logic to determine whether to declare Node as a runtime or build dep, whether to create a run hook or not, and which npm commands to run to build and start. Is that right?

Reply:
I was imagining splitting the scaffolding - and having to support two different workflows.  That said, if we are willing to accept some need for additional configuration in a plan.sh file (rather than just specifying the scaffolding) - we can potentially keep it in one.  We would be trading some simplicity for flexibility - something I am willing to do, but want the input of the community beforehand.